cmd/unlockerd/fileprovider.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+const (
+	tlsCertKey  = "tls-cert-path"
+	passwordKey = "password-path"
+)
+
+var (
+	defaultDatadir     = btcutil.AppDataDir("tdex-daemon", false)
+	defaultTLSCertPath = filepath.Join(defaultDatadir, "tls", "cert.pem")
+
+	tlsCertFlag = pflag.String(
+		tlsCertKey,
+		defaultTLSCertPath,
+		"the path of the TLS certificate file",
+	)
+	passwordFlag = pflag.String(
+		passwordKey,
+		"",
+		"the path of the file containing the password to unlock the wallet",
+	)
+)
+
+type fileProvider struct {
+	pwdPath     string
+	tlsCertPath string
+}
+
+func NewFileProvider() (provider, error) {
+	pwdPath := viper.GetString(passwordKey)
+	if pwdPath == "" {
+		return nil, fmt.Errorf("invalid flag: %s must not be null", passwordKey)
+	}
+	if !pathExists(pwdPath) {
+		return nil, fmt.Errorf("invalid flag: %s must be an existing path", passwordKey)
+	}
+
+	tlsCertPath := viper.GetString(tlsCertKey)
+	if tlsCertPath == "" {
+		return nil, fmt.Errorf("invalid flag: %s must not be null", tlsCertKey)
+	}
+
+	return &fileProvider{pwdPath, tlsCertPath}, nil
+}
+
+func (fp *fileProvider) Password() ([]byte, error) {
+	return ioutil.ReadFile(fp.pwdPath)
```

Comment:
ok maybe worht adding a README where we document the file format. it looks like we assume the file will just have the password there in it and we read all until EOF?

Reply:
exactly, the file should contain ONLY the password and nothing else. Maybe we could add a section dedicated to build&use the unlocker binary to the main README just like we did for the CLI.
Then, its detailed usage could be documented into dev-portal?